### PR TITLE
Adding alias type struct and map

### PR DESCRIPTION
**Pull request by handstuyennn**
Hi @Mytherin 

From [this thread](https://github.com/duckdb/duckdb/pull/3490).

I have fixed in Parser for Query:
`CREATE TYPE POINT AS STRUCT(x INT, y INT);`
and
`CREATE TYPE MAPPOINT AS MAP(INTEGER,INTEGER);`

I added tests for above too.

Please review

### CODE REVIEW

**Review by Mytherin — APPROVED** [review]
Thanks for the PR! Looks good. Some minor comments:

File: third_party/libpg_query/grammar/statements/select.y [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ GenericType:
 			// 	}
 		;
 
-opt_type_modifiers: '(' expr_list_opt_comma	 ')'				{ $$ = $2; }
+opt_type_modifiers: '(' opt_expr_list_opt_comma	 ')'				{ $$ = $2; }
```
[the remark this answers is not in the document]
Comment (handstuyennn) [reply third_party/libpg_query/grammar/statements/select.y:1336]: Duckdb doesn't allow `DECIMAL()` but it allows `ENUM()`.
I got an error when parsing for `ENUM()`. Could you give me your advise

[the remark this answers is not in the document]
Comment (Mytherin) [reply third_party/libpg_query/grammar/statements/select.y:1336]: That's fair. Could you then at least add tests for other scenarios where `opt_type_modifiers` is used to ensure that this does not cause other problems (i.e. `DECIMAL`, `DEC`, `NUMERIC`)?

**Comment by handstuyennn** [comment]
Hi @Mytherin 

I have just updated this branch and update new tests for `DECIMAL`, `DEC`, `NUMERIC`.
Please review

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**